mmdet/core/utils/dist_utils.py

```diff
@@ -42,15 +43,10 @@ def allreduce_grads(params, coalesce=True, bucket_size_mb=-1):
 
 
 class DistOptimizerHook(OptimizerHook):
+    """Deprecated optimizer hook for distributed training"""
 
-    def __init__(self, grad_clip=None, coalesce=True, bucket_size_mb=-1):
-        self.grad_clip = grad_clip
-        self.coalesce = coalesce
-        self.bucket_size_mb = bucket_size_mb
-
-    def after_train_iter(self, runner):
-        runner.optimizer.zero_grad()
-        runner.outputs['loss'].backward()
-        if self.grad_clip is not None:
-            self.clip_grads(runner.model.parameters())
-        runner.optimizer.step()
+    def __init__(self, *args, **kwargs):
+        warnings.warn(
+            '"DistOptimizerHook" is deprecated, please switch to'
+            '"mmcv.runner.OptimizerHook".', DeprecationWarning)
```

Comment:
`DeprecationWarning` is ignored by default, we may just use a `UserWarning` by leaving the second argument empty.